<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1301</width>
    <height>634</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>X0 Development Kit</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QTextEdit" name="textEdit">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>30</y>
      <width>391</width>
      <height>541</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>0</y>
      <width>81</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>x0 Code</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>470</x>
      <y>0</y>
      <width>81</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>VM ASM</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="inputEdit">
    <property name="geometry">
     <rect>
      <x>700</x>
      <y>30</y>
      <width>341</width>
      <height>91</height>
     </rect>
    </property>
    <property name="readOnly">
     <bool>false</bool>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>710</x>
      <y>0</y>
      <width>81</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Input</string>
    </property>
   </widget>
   <widget class="QPushButton" name="runBtn">
    <property name="geometry">
     <rect>
      <x>700</x>
      <y>130</y>
      <width>106</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string>Run</string>
    </property>
   </widget>
   <widget class="QPushButton" name="stepBtn">
    <property name="geometry">
     <rect>
      <x>840</x>
      <y>130</y>
      <width>106</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string>Step</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="outputEdit">
    <property name="geometry">
     <rect>
      <x>700</x>
      <y>210</y>
      <width>341</width>
      <height>91</height>
     </rect>
    </property>
    <property name="readOnly">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>710</x>
      <y>180</y>
      <width>81</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Output</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_5">
    <property name="geometry">
     <rect>
      <x>710</x>
      <y>320</y>
      <width>81</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Info</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="infoEdit">
    <property name="geometry">
     <rect>
      <x>700</x>
      <y>350</y>
      <width>581</width>
      <height>221</height>
     </rect>
    </property>
    <property name="readOnly">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QListWidget" name="asmList">
    <property name="geometry">
     <rect>
      <x>460</x>
      <y>30</y>
      <width>201</width>
      <height>501</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_6">
    <property name="geometry">
     <rect>
      <x>1090</x>
      <y>0</y>
      <width>81</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Stack</string>
    </property>
   </widget>
   <widget class="QListWidget" name="stackList">
    <property name="geometry">
     <rect>
      <x>1080</x>
      <y>30</y>
      <width>201</width>
      <height>271</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="compileBtn">
    <property name="geometry">
     <rect>
      <x>460</x>
      <y>540</y>
      <width>106</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string>Compile</string>
    </property>
   </widget>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1301</width>
     <height>32</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionLoad"/>
    <addaction name="actionSaveX0"/>
    <addaction name="actionSaveVM"/>
   </widget>
   <widget class="QMenu" name="menuBuild">
    <property name="title">
     <string>Build</string>
    </property>
    <addaction name="actionCompile"/>
    <addaction name="actionRun"/>
    <addaction name="actionSetPath"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuBuild"/>
  </widget>
  <action name="actionLoad">
   <property name="text">
    <string>Load</string>
   </property>
  </action>
  <action name="actionSaveX0">
   <property name="text">
    <string>Save x0 File</string>
   </property>
  </action>
  <action name="actionCompile">
   <property name="text">
    <string>Compile</string>
   </property>
  </action>
  <action name="actionRun">
   <property name="text">
    <string>Run</string>
   </property>
  </action>
  <action name="actionSaveVM">
   <property name="text">
    <string>Save VM Code</string>
   </property>
  </action>
  <action name="actionSetPath">
   <property name="text">
    <string>Set Compiler Path...</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>menubar</sender>
   <signal>triggered(QAction*)</signal>
   <receiver>MainWindow</receiver>
   <slot>handleActions(QAction*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>47</x>
     <y>11</y>
    </hint>
    <hint type="destinationlabel">
     <x>527</x>
     <y>169</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>compileBtn</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>compile()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>498</x>
     <y>587</y>
    </hint>
    <hint type="destinationlabel">
     <x>431</x>
     <y>366</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>runBtn</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>runASM()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>774</x>
     <y>183</y>
    </hint>
    <hint type="destinationlabel">
     <x>678</x>
     <y>249</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>stepBtn</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>step()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>897</x>
     <y>177</y>
    </hint>
    <hint type="destinationlabel">
     <x>681</x>
     <y>357</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>handleActions(QAction*)</slot>
  <slot>compile()</slot>
  <slot>runASM()</slot>
  <slot>step()</slot>
 </slots>
</ui>
